sonic_db_config/db_config.go

```diff
@@ -69,94 +73,60 @@ func GetDbNamespaceFromTarget(target string) (string, bool) {
 	}
 	return "", false
 }
-func GetDbList(ns string) map[string]interface{} {
-	if !sonic_db_init {
-		DbInit()
-	}
-	db_list, ok := sonic_db_config[ns]["DATABASES"].(map[string]interface{})
-	if !ok {
-		panic(fmt.Errorf("DATABASES' is not valid key in database_config.json file for namespace `%v` !", ns))
-	}
-	return db_list
-}
 
-func GetDbInst(db_name string, ns string) map[string]interface{} {
+func GetDbList(ns string) []string {
 	if !sonic_db_init {
 		DbInit()
 	}
-	db, ok := sonic_db_config[ns]["DATABASES"].(map[string]interface{})[db_name]
-	if !ok {
-		panic(fmt.Errorf("database name '%v' is not valid in database_config.json file for namespace `%v`!", db_name, ns))
-	}
-	inst_name, ok := db.(map[string]interface{})["instance"]
-	if !ok {
-		panic(fmt.Errorf("'instance' is not a valid field in database_config.json file for namespace `%v`!", ns))
+	db_vec := swsscommon.SonicDBConfigGetDbList()
+	// Translate from vector to array
+	length := int(db_vec.Size())
+	var db_list []string
+	for i := 0; i < length; i += 1 {
+		ns := db_vec.Get(i)
+		db_list = append(db_list, ns)
 	}
-	inst, ok := sonic_db_config[ns]["INSTANCES"].(map[string]interface{})[inst_name.(string)]
-	if !ok {
-		panic(fmt.Errorf("instance name '%v' is not valid in database_config.json file for namespace `%v`!", inst_name, ns))
-	}
-	return inst.(map[string]interface{})
+	return db_list
 }
 
 func GetDbSeparator(db_name string, ns string) string {
 	if !sonic_db_init {
 		DbInit()
 	}
-	db_list := GetDbList(ns)
-	separator, ok := db_list[db_name].(map[string]interface{})["separator"]
-	if !ok {
-		panic(fmt.Errorf("'separator' is not a valid field in database_config.json file!"))
-	}
-	return separator.(string)
+	separator := swsscommon.SonicDBConfigGetSeparator(db_name, ns)
+	return separator
 }
 
 func GetDbId(db_name string, ns string) int {
 	if !sonic_db_init {
```

Comment:
>sonic_db_init [](http://example.com/codeflow?start=5&length=13)

swsscommon already has this check and init.

Reply:
If this block removed, there is no value to define this function. We can remove the caller code to swsscommon function.

Reply:
https://github.com/sonic-net/sonic-swss-common/blob/master/common/dbconnector.cpp#L218-L227
You are right, we don't need dbinit here.